README.md

```diff
@@ -17,11 +32,438 @@ Xcode 16 or later.
+realtimeOptions.key = "<API_KEY>"
+realtimeOptions.clientId = "<clientId>"
+let realtime = ARTRealtime(options: realtimeOptions)
+let chatClient = DefaultChatClient(realtime: realtime, clientOptions: nil)
+```
+
+You can use [basic authentication](https://ably.com/docs/auth/basic) i.e. the API Key directly for testing purposes,
+however it is strongly recommended that you use [token authentication](https://ably.com/docs/auth/token) in production
+environments.
+
+To use Chat you must also set a [`clientId`](https://ably.com/docs/auth/identified-clients) so that users are
+identifiable.
+
+## Connections
+
+The Chat SDK uses a single connection to Ably, which is exposed via the `ChatClient#connection` property. You can use this
+property to observe the connection state and take action accordingly.
+
+### Current connection status
+
+You can view the current connection status at any time:
+
+```swift
+let status = await chatClient.connection.status
+let error = await chatClient.connection.error
+```
+
+### Subscribing to connection status changes
+
+You can subscribe to connection status changes by registering a listener, like so:
+
+```swift
+let subscription = chatClient.connection.onStatusChange(bufferingPolicy: .unbounded)
+for await statusChange in subscription {
+    print("Connection status changed to: \(statusChange.current)")
+}
+```
+
+To stop listening to changes, call the `finish` method on the returned subscription instance:
+
+```swift
+subscription.finish()
+```
+
+## Chat rooms
+
+### Creating or retrieving a chat room
+
+You can create or retrieve a chat room with name `"basketball-stream"` this way:
+
+```swift
+let room = try await chatClient.rooms.get(roomID: "basketball-stream", options: RoomOptions())
+```
+
+The second argument to `rooms.get` is a `RoomOptions` argument, which tells the Chat SDK what features you would like your room to use and
+how they should be configured.
+
+You can also use `RoomOptions()` to enable all room features with the default configuration.
+
+For example, you can set the timeout between keystrokes for typing events as part of the room options. Sensible defaults for each of the
+features are provided for your convenience:
+
+- A typing timeout (time of inactivity before typing stops) of 5 seconds.
+- Entry and subscription to presence.
+
+Here’s an example demonstrating how to specify a custom typing timeout of 3 seconds:
+
+```swift
+let room = try await chatClient.rooms.get(roomID: "basketball-stream",
+                                          options: .init(typing: TypingOptions(timeout: 3.0)))
+```
+
+In order to use the same room but with different options, you must first `release` the room before requesting an instance with the changed
+options (see below for more information on releasing rooms).
+
+Note that:
+
+- If a `release` call is currently in progress for the room (see below), then a call to `get` will wait for that to resolve before resolving
```

Comment:
`reject` and `resolve` are JS promise terms; I’d suggest `fail` and `complete`.

Reply:
Updated